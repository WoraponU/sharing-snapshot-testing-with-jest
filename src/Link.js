import React, { Component } from 'react'

class Link extends Component {
  render() {
    const { link } = this.props
    return (
      <div>
        link: <a href={link} target="_blank">{link}</a>
      </div>
    )
  }
}

export default Link
