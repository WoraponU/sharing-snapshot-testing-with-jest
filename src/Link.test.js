import React from 'react'
import { configure, shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'
import shallowRenderer from 'react-test-renderer/shallow'
import Adapter from 'enzyme-adapter-react-15';
import Link from './Link'

configure({ adapter: new Adapter() });

describe('Test link component', () => {
  it('shallow link props', () => {
    const wrapper = shallow(<Link link="www.fackbook.com" />);
    expect(wrapper.contains(<a href="www.fackbook.com" target="_blank">www.fackbook.com</a>)).toBe(true);
  })

  it('shallow snapshot', () => {
    const renderer = new shallowRenderer
    const snapshot = renderer.render(<Link link="www.fackbook.com"/>)
    expect(snapshot).toMatchSnapshot()
  })

  it('full snapshot', () => {
    const html = renderer.create(<Link link="www.fackbook.com"/>).toJSON()
    expect(html).toMatchSnapshot()
  })
})