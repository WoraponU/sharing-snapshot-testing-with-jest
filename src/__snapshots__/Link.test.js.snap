// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test link component full snapshot 1`] = `
<div>
  link: 
  <a
    href="www.fackbook.com"
    target="_blank"
  >
    www.fackbook.com
  </a>
</div>
`;

exports[`Test link component shallow snapshot 1`] = `
<div>
  link: 
  <a
    href="www.fackbook.com"
    target="_blank"
  >
    www.fackbook.com
  </a>
</div>
`;
